src/o365/spo/commands/list/list-sitescript-get.ts

```diff
@@ -0,0 +1,240 @@
+import auth from '../../SpoAuth';
+import config from '../../../../config';
+import commands from '../../commands';
+import GlobalOptions from '../../../../GlobalOptions';
+import * as request from 'request-promise-native';
+import {
+  CommandOption,
+  CommandValidate,
+  CommandError
+} from '../../../../Command';
+import SpoCommand from '../../SpoCommand';
+import Utils from '../../../../Utils';
+import { Auth } from '../../../../Auth';
+import { ContextInfo, 
+  // ClientSvcResponse, 
+  // ClientSvcResponseContents 
+} from '../../spo';
+import { ListInstance } from './ListInstance';
+
+const vorpal: Vorpal = require('../../../../vorpal-init');
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+}
+
+class SpoListSitescriptGetCommand extends SpoCommand {
+  public get name(): string {
+    return commands.LIST_SITESCRIPT_GET;
+  }
+
+  public get description(): string {
+    return 'Extracts a site script from a SharePoint list';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = (!(!args.options.listId)).toString();
+    telemetryProps.listTitle = (!(!args.options.listTitle)).toString();
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    const resource: string = Auth.getResourceFromUrl(args.options.webUrl);
+    let formDigest: string;
+
+    if (this.debug) {
+      cmd.log(`Retrieving access token for ${resource}...`);
+    }
+
+    auth
+      .ensureAccessToken(auth.service.resource, cmd, this.debug)
```

Comment:
Because user specifies the URL of the site, we should use `getAccessToken` passing in the resource we retrieved in line 49. In the current setup, the command fails when you're connected to the tenant admin site

Reply:
Hi @waldekmastykarz, I tried to use `getAccessToken` instead it works when logging in to a specific site but it is still failing when logging in to the tenant admin. Do you think about anything else I should look at ?

Reply:
Just realized I broke the tests, let me look at it :)